modin/backends/pandas/query_compiler.py

```diff
@@ -1828,6 +1832,56 @@ def applyier(df, internal_indices, other=[], internal_other_indices=[]):
     # END Map across rows/columns
 
     # __getitem__ methods
+    def getitem_array(self, key):
+        """
+        Get column or row data specified by key.
+
+        Parameters
+        ----------
+        key : Series, numpy.ndarray, pandas.Index or list
+            Target numeric indices or labels by which to retrieve data.
+
+        Returns
+        -------
+        BaseQueryCompiler
+            A new Query Compiler.
+        """
+        # TODO: dont convert to pandas for array indexing
+        if isinstance(key, type(self)):
+            key = key.to_pandas()
+        if is_bool_indexer(key):
+            if isinstance(key, pandas.Series) and not key.index.equals(self.index):
```

Comment:
`key` will never be a `pandas.Series` here because the former is converted to `pandas.DataFrame` above. When we convert `key` to `Series` in API layer, we know what object will be after conversion in depend of preliminary condition `if isinstance(key, Series)`.